kbatch/kbatch/_core.py

```diff
@@ -340,7 +340,7 @@ def duration(job) -> str:
     if job["status"]["succeeded"]:
         end_time = datetime.datetime.fromisoformat(job["status"]["completion_time"])
     elif job["status"]["failed"]:
-        end_time = None
+        end_time = datetime.datetime.fromisoformat(job["status"]["conditions"][0]["last_transition_time"])
```

Comment:
Nice! I think we might want a little more guarding to make sure we get the right value and don't get an error if information is missing, or other conditions might be present.

How about:

- iterate over conditions
- pick the first condition where `condition["type"] == "Failed"` and use its `last_transition_time` (alternately, pick the _latest_ transition time after looking at all conditions, but I'm not sure there can be more than one Failed condition)
- if no matching condition is found with type=Failed, use None

which would look like:

```suggestion
        end_time = None
        for condition in job["status"]["conditions"]:
            if condition["type"] == "Failed":
                end_time = datetime.datetime.fromisoformat(condition["last_transition_time"])
                break
```

Reply:
Hi,
Thanks for the suggestion! I willingly made the commit quite "raw", without additional guarding since I did not know the best way to handle it.

Iterating over the conditions is elegant, but I would suggest to accept your changes as they are (i.e. with `break`) since that it seems that there could be only one with `(type, status)` = `(Failed, True)` ([ref here](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1JobStatus.md)).